platform/security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java

```diff
@@ -445,23 +449,23 @@ private void createSecurityToken(SoapVersion version, SOAPFactory soapFactory,
         }
     }
 
-    private Subject getSubject() {
-        if (Security.tokenAboutToExpire(cachedAnonymousSubject)) {
+    private synchronized Subject getSubject(String ipAddress) {
+        Subject subject = cachedAnonymousSubjectMap.get(ipAddress);
+        if (Security.tokenAboutToExpire(subject)) {
             AnonymousAuthenticationToken token = new AnonymousAuthenticationToken(
-                    BaseAuthenticationToken.DEFAULT_REALM);
-            LOGGER.debug("Getting new Anonymous user token");
-            //synchronize the step of requesting the assertion, it is not thread safe
-            synchronized (lock) {
-                try {
-                    cachedAnonymousSubject = securityManager.getSubject(token);
-                } catch (SecurityServiceException sse) {
-                    LOGGER.warn("Unable to request subject for anonymous user.", sse);
-                }
+                    BaseAuthenticationToken.DEFAULT_REALM, ipAddress);
+            LOGGER.debug("Getting new Anonymous user token for {}", ipAddress);
+            try {
```

Comment:
It looks like we're missing the synchronized block here now. The securityManager.getSubject() call is still not thread safe.